---
title: "Independent Project Statistical Analysis"
author: "Gillyan Valencia"
output:
  pdf_document: default
  html_document: default
---
## Introduction

Thermophilic microorganisms, capable of living at extreme temperatures in habitats like geothermal springs, offer unique insights into the evolution of microbial life and life’s potential in harsh environments. (Damer and Deamer, 2020). While microbial evolution has been studied across different environments (Nguyen et al., 2021), the evolution of population genomics and community-level dynamics within geothermal springs remain poorly understood. The Great Boiling Spring (GBS) in Gerlach, Nevada, provides an ideal system to explore these dynamics, as its microbial communities are structured by temperature gradients and micro-environments such as sediment and the water column (Cole et al., 2013). Thermoflexus hugenholtzii—a dominant thermophile across sites in the Great Boiling Spring (GBS) system in Gerlach, Nevada—provides an ideal model to investigate fine-scale genetic variation across environmental gradients. This study uses metagenomic sequencing and population genomics tools to examine the microevolutionary dynamics of T. hugenholtzii, focusing on single nucleotide polymorphisms (SNPs) and single codon variations (SCVs) within genes critical to core cellular functions. 

Given that temperature is a key factor shaping microbial communities in GBS (Cole et al., 2013; Kees et al., 2022), this study aims to examine how microbial community composition and population genomics evolve in response to seasonal temperature variations. A high-quality metagenome-assembled genome (MAG) generated from the most complete, least contaminated sample (June Site B) served as the reference genome. SNP frequencies across five sample sites were analyzed in relation to site-specific and seasonal temperature fluctuations. This approach allows for the exploration of how environmental variables influence genetic variation within microbial populations, contributing to a broader understanding of microbial genome plasticity and ecological adaptation in extreme ecosystems.

## Research Question

How do temperature and site-specific environmental conditions influence single nucleotide polymorphism (SNP) and single codon variation (SCV) in core functional genes of Thermoflexus hugenholtzii across spatial and seasonal gradients in Great Boiling Spring?

## Specific Aims

**Aim 1: **
To quantify and compare SNP frequencies in three conserved genes of T. hugenholtzii across five metagenomic samples and assess their correlation with temperature and sampling site.

**Aim 2:**
To quantify and compare SCV frequencies in three conserved genes of T. hugenholtzii across five metagenomic samples and assess their correlation with temperature and sampling site.

## Approach

I first checked the distribution of the data to figure out which statistical method was best. The data for all 3 genes of interest was not distributed normally. I used a generalized linear mixed model (GLMM) with a beta distribution to analyze single codon variants (SCVs) frequency, where the frequencies all fell between 0 and 1. The model included fixed effects for Site and Month their interactions on SCVs within the rpoB, recA, and gyrB genes. This allowed for me to assess both main effects and gene-specific environmental interactions. The model was fit using the glmmTMB package in R.

```{r}
# Load SCV dataset
scv <- read.csv("../Data/Thermo_Hugo_SCV.csv")
#str(scv)
```

```{r}
library(tidyverse)
```


```{r}
# Check Normality
rpoB_scv_data <- subset(scv, Gene_ID == "rpoB")
rpoB_scv_hist <- ggplot(rpoB_scv_data, aes(x = scv_freq)) +
  geom_histogram(binwidth = 0.01)
#rpoB_scv_hist

recA_scv_data <- subset(scv, Gene_ID == "recA")
recA_scv_hist <- ggplot(recA_scv_data, aes(x = scv_freq)) +
  geom_histogram(binwidth = 0.01)
#recA_scv_hist

gyrB_scv_data <- subset(scv, Gene_ID == "gyrB")
gyrB_scv_hist <- ggplot(gyrB_scv_data, aes(x = scv_freq)) +
  geom_histogram(binwidth = 0.01)
#gyrB_scv_hist
```

## Hypotheses
**Null hypothesis (H0):**
Site and Month have no effect on SCV frequency in any of the genes tested.

**Alternative hypothesis (HA):**
SCV frequency varies significantly by Site and/or Month for at least one gene.

```{r}
# SCV

# SCV Frequency GLM
SCV_GLMM <- glmmTMB(
  scv_freq ~ Gene_ID * Site + Gene_ID * Month,
  data = scv,
  family = beta_family()
)

# Add Model Predictions
scv$predicted <- predict(SCV_GLMM, type = "response")

# 3. Clean data
scv_clean <- scv %>%
  filter(!is.na(Month)) %>%
  mutate(
    Month = factor(trimws(as.character(Month))),  # Clean text and convert to factor
    SiteMonth = interaction(Site, Month, sep = "_")  # Create grouping variable
  )

# Order of x-axis
scv_clean$SiteMonth <- factor(scv_clean$SiteMonth,
  levels = c("A_February", "B_February", "A_June", "B_June", "C_June")
)
```

```{r Figure_1, fig.cap="Observed and predicted SCV frequencies by site and sampling month across three genes (gyrB, recA, and rpoB) in *Thermoflexus hugenholtzii*. Points represent individual observations. Black diamonds show model-predicted means from a GLMM with a beta distribution. Data are grouped by site and month and faceted by gene."}

# Plot
ggplot(scv_clean, aes(x = SiteMonth, y = scv_freq)) +
  geom_jitter(aes(color = Site), width = 0.2, alpha = 0.6) +
  stat_summary(aes(y = predicted), fun = mean, geom = "point",
               shape = 18, size = 3, color = "black") +
  facet_wrap(~ Gene_ID) +
  labs(
    title = "SCV Frequency Variance in Thermoflexus hugenholtzii",
    x = "Site and Month",
    y = "SCV Frequency",
    
  ) +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "none")
# Summary Statistics

summary(SCV_GLMM)
```

## References

Cole, J. K., Peacock, J. P., Dodsworth, J. A., Williams, A. J., Thompson, D. B., Dong, H., Wu, G., & Hed- lund, B. P. (2013). Sediment microbial communities in Great Boiling Spring are controlled by temperature and distinct from water communities. The ISME Journal, 7(4), 718– 729. https://doi.org/10.1038/ismej. 2012.157

Damer, B., & Deamer, D. (2020). The Hot Spring Hypothesis for an Origin of Life. Astrobiology, 20(4), 429–452. https://doi.org/10.1089/ast.2019.2045

Kees, E. D., Murugapiran, S. K., Bennett, A. C., & Hamilton, T. L. (2022). Distribution and Genomic Variation of Thermophilic Cyanobacteria in Diverse Microbial Mats at the Upper Temperature Limits of Photosynthesis. mSystems, 7(5), e00317-22. https://doi.org/10.1128/msystems.00317-22

Nguyen, J., Lara-Gutiérrez, J., & Stocker, R. (2021). Environmental fluctuations and their effects on mi- crobial communities, populations and individuals. FEMS Microbiology Reviews, 45(4), fuaa068. https: //doi.org/10.1093/femsre/fuaa068